generators/server/prompts.js

```diff
@@ -156,6 +156,32 @@ function askForServerSideOpts(meta) {
                 return `Could not find a valid JHipster UAA server in path "${input}"`;
             }
         },
+        {
+            // cache is mandatory for gateway and define later to 'hazelcast' value
+            when: response => applicationType !== 'gateway',
+            type: 'list',
+            name: 'cacheProvider',
+            message: 'Do you want to use spring cache?',
+            choices: [
+                {
+                    value: 'ehcache',
+                    name: 'Yes, with ehcache implementation (local cache, for a single node)'
+                },
+                {
+                    value: 'hazelcast',
+                    name: 'Yes, with HazelCast implementation(distributed cache, for multiple nodes)'
+                },
+                {
+                    value: 'infinispan',
+                    name: '[BETA] Yes, with Infinispan (hybrid cache, for multiple nodes)'
+                },
+                {
+                    value: 'no',
+                    name: 'No'
+                }
+            ],
+            default: 'no'
```

Comment:
To retain current behaviour do the below 
`default: (applicationType === 'microservice' || applicationType === 'uaa'| ) ? 1 : 0`